comp/core/workloadmeta/impl/component_mock.go

```diff
@@ -21,31 +21,33 @@ import (
 
 // Mock implements mock-specific methods.
 type Mock interface {
-	Component
+	wmdef.Component
 
 	// The following are for testing purposes and should maybe be revisited
 	// Set allows setting an entity in the workloadmeta store
-	Set(entity Entity)
+	Set(entity wmdef.Entity)
 
 	// Unset removes an entity from the workloadmeta store
-	Unset(entity Entity)
+	Unset(entity wmdef.Entity)
 
 	// GetConfig returns a Config Reader for the internal injected config
 	GetConfig() config.Reader
 
 	// GetConfig returns a Config Reader for the internal injected config
-	GetNotifiedEvents() []CollectorEvent
+	GetNotifiedEvents() []wmdef.CollectorEvent
 
 	// SubscribeToEvents returns a channel that receives events
-	SubscribeToEvents() chan CollectorEvent
+	SubscribeToEvents() chan wmdef.CollectorEvent
 }
 
 // MockModule defines the fx options for the mock component.
 func MockModule() fxutil.Module {
```

Comment:
This and the `MockModuleV2` function should be part of the `fx-mock` package 😄 Same we did for the replay component https://github.com/DataDog/datadog-agent/pull/25512